src/core/lib/layer-manager.js

```diff
@@ -658,8 +658,8 @@ export default class LayerManager {
     const radius = this._pickingRadius;
     const selectedInfos = this.pickObject({x: pos.x, y: pos.y, radius, mode: 'hover'});
 
-    const firstInfo = selectedInfos.find(info => info.index >= 0);
-    if (firstInfo && this._onLayerHover) {
+    const firstInfo = selectedInfos.find(info => info.index >= 0) || null;
```

Comment:
Maybe a comment about the `null`, 
```
// Note: API docs state that info must be `null`. 
```

And why is this only in the hover case, not in the click case?

Share code between the two cases?